packages/babel-parser/src/tokenizer/index.ts

```diff
@@ -195,6 +196,34 @@ export default abstract class Tokenizer extends CommentsParser {
     return this.input.charCodeAt(this.nextTokenStart());
   }
 
+  /**
+   * Similar to nextToken, but it will stop at line break when it is seen before the next token
+   *
+   * @returns {number} position of the next token start or line break, whichever is seen first.
+   * @memberof Tokenizer
+   */
+  nextTokenInLineStart(): number {
+    return this.nextTokenInLineStartSince(this.state.pos);
+  }
+
+  nextTokenInLineStartSince(pos: number): number {
+    skipWhiteSpaceInLine.lastIndex = pos;
+    return skipWhiteSpaceInLine.test(this.input)
+      ? skipWhiteSpaceInLine.lastIndex
+      : pos;
+  }
```

Comment:
```
nextTokenInLineStart(pos = this.state.pos): number {
```
Would this be better?

Reply:
default parameter adds a new branching. I can inline `nextTokenInLineStartSince` since we are not using it anyway. (An optimizing compiler will inline it for sure).

Reply:
Ah, `nextTokenInLineStartSince` will be used in parsing `await using foo = bar()`, when we do a second lookahead from `using`.